File: daily_python/2425_Bitwise_XOR_of_All_Pairings.py
import os
import sys
from typing import List

class Solution:
    def xorAllNums(self, nums1: List[int], nums2: List[int]) -> int:
        ans = 0
        if len(nums2) % 2 == 1:
            for i in nums1:
                ans ^= i
        if len(nums1) % 2 == 1:
            for i in nums2:
                ans ^= i
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums1 = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            nums2 = list(map(int, lines[1].split('=')[1].strip()[1:-2].split(',')))
            dataset.append((nums1, nums2))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums1, nums2 in dataset:
        results.append(solution.xorAllNums(nums1, nums2))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2025/2425_Bitwise_XOR_of_All_Pairings.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int ans = 0;
        if (nums2.size() % 2){
            for (int i : nums1){
                ans ^= i;
            }
        }
        if (nums1.size() % 2){
            for (int i : nums2){
                ans ^= i;
            }
        }
        return ans;
    }
};

int main(){
    vector<int> nums1 = {2,1,3};
    vector<int> nums2 = {10,2,5,0};
    // vector<int> nums1 = {1,2};
    // vector<int> nums2 = {3,4};
    Solution S;
    cout << S.xorAllNums(nums1, nums2) << endl;
    return 0;
}